### ISSUE TITLE
[BUG] Volumes don't start replication and stay degraded until all stopped replicas are deleted manually

### ISSUE THREAD

**Issue by Funzinator** [longhorn/longhorn#5417]
## Describe the bug (🐛 if you encounter this issue)

Every day around 3:30 am, some of my volumes become degraded. I assume it's because of daily backup jobs that put pressure to the IO. At least it helps to reproduce the issue ;-).
The problem is, that the replicas don't replicate anymore. They're all stuck. I have then to go through all volumes and delete the stopped replicas. Once the last stopped replica is deleted, the replication starts. I first assumed that I might not have enough space available but added another disk recently without any improvement.

## To Reproduce

In my case, it happens every evening. But I don't know how to replicate that on a fresh cluster.

## Expected behavior

I expect the replication to start and the volumes to be healthy.

## Log or Support bundle

[supportbundle_71c8e882-4ef9-4984-8e38-30492428c419_2023-02-21T07-50-04Z.zip](https://github.com/longhorn/longhorn/files/10791075/supportbundle_71c8e882-4ef9-4984-8e38-30492428c419_2023-02-21T07-50-04Z.zip)

## Environment

 - Longhorn version: 1.4.0
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm (`enablePSP: true`)
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: k3s v1.24.10+k3s1 
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 6
 - Node config
   - OS type and version: Rocky Linux 8.7 (for all except 1 worker that is still on Raspbian 11.6)
   - CPU per node: 5 Raspberry Pi 4B with each 4 cores, 1 Node with 2 Cores, 1 Node with 4 Cores
   - Memory per node: 4 Raspberry Pi 4B with 8 GB, 1 Raspberry Pi 4B with 4 GB (master), 2 Nodes with 16 GB
   - Disk type(e.g. SSD/NVMe): 1 SSD (via SATA), 1 SSD (via USB), 3 NVMe (via USB)
   - Network bandwidth between the nodes: 1 GBit/s, same switch
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster: 29

## Additional context

I have the feeling that it started after the upgrade to 1.4.0, but I am not sure.

Thanks in advance for any pointer,
Funzi

**Comment by Funzinator** [comment]
Maybe that helps: It looks in the UI like that. 

<img width="1726" alt="Screenshot 2023-02-21 at 09 06 59" src="https://user-images.githubusercontent.com/8009150/220285391-fa70d5fb-08f0-4f89-8adc-2ad2e18dbdab.png">

It usually helps if I delete the N/A replicas to get the replication going. I am also wondering why the replica on the right states "Running" but is formatted in black instead of blue

**Comment by weizhe0422** [comment]
The N/A replicas: `pvc-36ccafc6-8b77-4280-891f-3bee2569b9fc-r-77ee3bf7` seem to be unable to find a node that satisfies the size.
```txt
2023-02-21T08:50:27.215996173+01:00 time="2023-02-21T07:50:27Z" level=error msg="There's no available disk for replica pvc-36ccafc6-8b77-4280-891f-3bee2569b9fc-r-77ee3bf7, size 1073741824"
2023-02-21T08:50:27.216170411+01:00 time="2023-02-21T07:50:27Z" level=error msg="unable to schedule replica" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=rpi4-8g-2 owner=rpi4-8g-2 replica=pvc-36ccafc6-8b77-4280-891f-3bee2569b9fc-r-77ee3bf7 state=attached volume=pvc-36ccafc6-8b77-4280-891f-3bee2569b9fc
```

And the replica: `pvc-36ccafc6-8b77-4280-891f-3bee2569b9fc-r-ddaccc84` is under rebuilding but waiting the other replica on the same node finish rebuilding. 
```
2023-02-21T08:50:00.679759196+01:00 time="2023-02-21T07:50:00Z" level=debug msg="Replica rebuildings for map[pvc-36ccafc6-8b77-4280-891f-3bee2569b9fc-r-ddaccc84:{} pvc-cbb7937b-23e4-4ab0-b398-38282a9f5e69-r-188e60e9:{}] are in progress on this node, which reaches or exceeds the concurrent limit value 2" controller=longhorn-replica dataPath= node=x220-1 nodeID=x220-1 ownerID=x220-1 replica=pvc-ea25398e-4b0d-489b-8a44-2dc207a65ab1-r-af50c36c
```

I guess that's why after delete some N/A replicas make normal.

**Comment by Funzinator** [comment]
Thanks, I have indeed missed that.

What puzzles me `pvc-36ccafc6-8b77-4280-891f-3bee2569b9fc` is 1 GiB in size. That's one of the smallest volumes in use and there is definitely enough space to fit another GiB. I have almost 2 TiB free across all nodes.

The rest sounds like some sort of deadlock to me then and increasing the concurrent replication limit might treat the symptom but not the cause.

Can I increase the log level more? In about 6 hours, it will most likely happen again ;-)

**Comment by weizhe0422** [comment]
I'm trying to figure out why there are more than 3 copies of the same volume.

Can you check the node status include: `x220-1`, `rpi4-8g-1` and `rpi4-8g-3` to see if they are schedulable?

Since I found these three nodes seems not stable, and it makes replicas on the nodes evicted.
![Image](https://user-images.githubusercontent.com/17548901/220668949-d7ada0c6-7db0-4bb6-b065-df0335ea1460.png)

I thought this might have made the system try to rebuild a new replica on another node and then there was a conflict and exhausted resources.
```txt
2023-02-21T06:09:41.869408289+01:00 time="2023-02-21T05:09:41Z" level=debug msg="Node x220-1 is not ready or schedulable, replica pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e-r-c5e8075d could get evicted" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=rpi4-8g-2 owner=rpi4-8g-2 replicaAutoBalanceType=node state=attached volume=pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e
2023-02-21T06:09:41.869700211+01:00 time="2023-02-21T05:09:41Z" level=debug msg="Cannot use node rpi4-gold, does not allow scheduling" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=rpi4-8g-2 owner=rpi4-8g-2 replicaAutoBalanceType=node state=attached volume=pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e
2023-02-21T06:09:41.870530812+01:00 time="2023-02-21T05:09:41Z" level=debug msg="Found 1 node available for auto-balance duplicates in map[rpi4-8g-1:[] rpi4-8g-3:[]]" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=rpi4-8g-2 owner=rpi4-8g-2 replicaAutoBalanceType=node state=attached volume=pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e
2023-02-21T06:09:41.870793327+01:00 time="2023-02-21T05:09:41Z" level=debug msg="Found 1 replica candidate for auto-balance" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=rpi4-8g-2 owner=rpi4-8g-2 replicaAutoBalanceOption=least-effort state=attached volume=pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e
2023-02-21T06:09:41.871302524+01:00 time="2023-02-21T05:09:41Z" level=info msg="Cannot find a reusable failed replicas for volume pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e"
2023-02-21T06:09:41.871570835+01:00 time="2023-02-21T05:09:41Z" level=debug msg="A new replica pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e-r-07479fda will be replenished during rebuilding" accessMode=rwo controller=longhorn-volume frontend=blockdev migratable=false node=rpi4-8g-2 owner=rpi4-8g-2 state=attached volume=pvc-36335e5c-bad1-4fcd-be4f-c64dd3efd29e
```

**Comment by Funzinator** [comment]
Of course, tonight was the first time in weeks that it recovered itself. I still had degraded volumes but after a few hours, it was all healthy again without manual intervention.

However, whenever I checked in the monitoring of the last 7 days. The only time I had not all 5 nodes schedulable were reboots I initiated myself (and like 8 hours away from the stucked state):
<img width="1317" alt="Screenshot 2023-02-22 at 21 34 46" src="https://user-images.githubusercontent.com/8009150/220753018-e97eb944-8153-4b4e-a936-f0fabbacc2eb.png">

So according to the Prometheus metrics, the nodes were schedulable all the time.

Could it be that the node runs out of disk space due to the many concurrent and unsuccessful tries to create replicas there? However, the metrics don't complain here as well and max. usage of storage is below 50 % overall:
<img width="1042" alt="Screenshot 2023-02-22 at 21 42 15" src="https://user-images.githubusercontent.com/8009150/220754087-987abd6b-5a1d-43fe-8a77-e99cc25f95d4.png">
